Gordon360/Services/HousingService.cs

```diff
@@ -135,6 +135,31 @@ public AA_ApartmentHalls[] GetAllApartmentHalls()
             return result;
         }
 
+        /// <summary>
+        /// Calls a stored procedure that tries to get the editor ID of a given application ID
+        /// </summary>
+        /// <param name="applicationID"> The application ID for which the editor ID would be </param>
+        /// <returns>
+        /// The id of the editor or
+        /// null if the user is a member but not an editor of a given application
+        /// </returns>
+        public string GetEditorID(int? applicationID)
```

Comment:
Is there a particular reason why this param is nullable?

Reply:
If it is because the return type of the `GetApplicationID` is nullable, and you need to pass that return value into this method, then I would suggest that you make this param non-nullable int and perform a conversion in the place that calls this method. ~~I believe I do this exact thing inside the `GetAllApplications` method, which is near the bottom of this `HousingService.cs` file~~ (sorry, it is in a branch I have in progress). 

I believe it was as simple as 
```
var editorID = housingService.GetEditorID(applicationID.Value);
```
though 
```
var editorID = housingService.GetEditorID((int)applicationID);
``` 
should work as well

Reply:
Yeah, you're right. I did it this way because of its return type. Wasn't thinking too much about it when I wrote it haha. And I agree that it makes more sense to have the param as non-nullable int.

Reply:
For nullable int (and other nullables as well) you can use `if (applicationID.HasValue)` instead of `if (applicationID != null)` and then use `applicationID.Value` to get the non-nullable version, basically.
(I think there may be efficiency reasons as to why the `variable.HasValue` is preferred over `variable != null`, but I haven't looked into that too hard. The `HasValue` just seems to work better when intentionally using nullable variants of an existing type, based on what I've seen so far.